### PR TITLE
WIP: event payload format

### PR DESCRIPTION
**Pull request by mlagally**
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/wot-profile/pull/198.html" title="Last updated on Jun 28, 2022, 6:02 PM UTC (7346066)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wot-profile/198/b0ebae6...7346066.html" title="Last updated on Jun 28, 2022, 6:02 PM UTC (7346066)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -3053,15 +3053,105 @@ <h2>Identifier</h2>
...
+                A WebThing MAY resend the same event with the same <code>id</code>. 
+                If an event is resent with the same <code>id</code>, a consumer SHOULD treat it as
+                duplicate.</td>
+            </tr>
+            <tr>
+              <td>source</td>
+              <td>uri</td>
+              <td>mandatory, non-empty</td>
+              <td>The combination of <code>source</code> + <code>id</code> MUST be unique for each distinct event. 
+                The source MUST be set to the <code>baseURI</code> of the WebThing.</td>
```
Comment (egekorkan) [inline index.html:3112]: what is `baseURI`? Why do we need this if we have TDs?

File: index.html [hunk 2]
```diff
@@ -3053,15 +3053,105 @@ <h2>Identifier</h2>
...
+            <tr>
+              <td>data</td>
+              <td>string</td>
+              <td>optional</td>
+              <td>JSON encoded <code>data</code> object of the event affordance.</td>
+            </tr>
+          </tbody>
+        </table>
+        <span class="rfc2119-assertion" id="profile-event-payload-size">
+          The size of an event object MUST NOT exceed 64KB, to ensure interoperability with all kinds of consumers.
```
Comment (egekorkan) [inline index.html:3153]: where does this come from?

**Review by egekorkan — COMMENTED** [review]
Even though it is a WIP, I had a look. I find a fundamental problem with using cloud events in a profile that is meant for greenfield applications. It adds too much metadata to payload and does not make sense in a WoT specification.

File: index.html [hunk 3]
```diff
@@ -3053,15 +3053,105 @@ <h2>Identifier</h2>
...
+            </tr>
+            <tr>
+              <td>time</td>
+              <td>string</td>
+              <td>optional, compliant with [[RFC3339]]</td>
+              <td>time when the event occurred. If the WebThing does not have a clock, it MUST use the same
+                algorithm on all event affordances to determine the value of this field.
+            </tr>
+            <tr>
+              <td>datacontenttype</td>
```
Comment (sebastiankb) [inline index.html:3139]: why do we need 'datacontenttype'. forms already have a 'contentType' defined

File: index.html [hunk 4]
```diff
@@ -3053,15 +3053,105 @@ <h2>Identifier</h2>
...
+              <td>string</td>
+              <td>optional, [[RFC2046]] compliant MIME type, when set the value MUST be "<code>application/json</code>"</td>
+              <td>If a WebThing contains a <code>data member</code>, this field MUST be present.</td>
+            </tr>
+            <tr>
+              <td>data</td>
+              <td>string</td>
```
Comment (sebastiankb) [inline index.html:3146]: why do we have a `string` here?

File: index.html [hunk 5]
```diff
@@ -3053,15 +3053,105 @@ <h2>Identifier</h2>
...
+          <tbody>
+            <tr>
+              <td>specversion</td>
+              <td>string</td>
+              <td>mandatory, non-empty</td>
+              <td>The <code>specversion</code> MUST have the value "1.0"</td>
+            </tr>
+            <tr>
+              <td>id</td>
+              <td>string</td>
```
Comment (sebastiankb) [inline index.html:3100]: TD uses anyURI

File: index.html (hunk 2 above)
Comment (mlagally) [reply index.html:3153]: A pragmatic limit. 
Why would we have more than 64k for an event payload?

We should discuss the limits together.

Comment (egekorkan) [reply index.html:3153]: Similarly, why would need more than 32KB? Or 35KB?

File: index.html (hunk 1 above)
Comment (mlagally) [reply index.html:3832]: It is a URI in the original sense, i.e. an indentifier for a unique resource (= the thing)

File: index.html (hunk 2 above)
Comment (mlagally) [reply index.html:3873]: 64K can be addressed by 16 bits, which is still a limit on some platforms.

File: index.html (hunk 5 above)
Comment (mlagally) [reply index.html:3820]: good point - we could restrict the string contents to URI formats only.

**Comment by mlagally** [comment]
Won't merge - superseeded by PR 235

File: index.html (hunk 4 above)
Comment (sebastiankb) [reply index.html:3866]: image

I think, it should be definitely any type.

**Pull request closed without merging**